File: src/main/java/com/paqi/friendsystem/service/LiteratureService.java
package com.paqi.friendsystem.service;

import com.paqi.friendsystem.entity.Literature;

import java.util.ArrayList;

/**
 * @author PQ
 * @Description 文章服务类
 * @Date 下午6:31 17/3/2020
 * @version 3.2.0
**/
public interface LiteratureService {
    /**
     * 添加一篇文章
     * @author PQ
     * @Description 填一篇文章
     * @param literature：文章
     * @return 返回该文章的id
     * @Date 下午6:36 17/3/2020
     * @version 3.2.0
    **/
    int addLiterature(Literature literature);

    /**
     * 删除一篇文章
     * @author PQ
     * @Description 删除一篇文章
     * @param literatureId：文章id
     * @return 若删除返回true，否则返回false
     * @Date 下午9:26 17/3/2020
     * @version 3.3.0
    **/
    boolean deleteLiterature(int literatureId);

    /**
     * 根据部落id查询该部落所有文章
     * @author PQ
     * @Description 查询某兴趣全部文章
     * @param funId：文章id
     * @return 文章列表
     * @Date 13:45 20/3/2020
     * @version 3.4.13
    **/
    ArrayList<Literature> getAllLiteratureInFun(int funId);

    /**
     * 根据文章状态返回文章列表
     * @author PQ
     * @Description 文章内容状态
     * @return 文章列列表
     * @Date 23:39 20/3/2020
     * @version 3.4.19
    **/
    ArrayList<Literature> getLiteratureByStatus(int status);

    /**
     * @author PQ
     * @Description 3.4.19
     * @param literatureId:文章id
     * @param status: 文章状态
     * @return 更改成功返回true，否则返回false
     * @Date 00:48 21/3/2020
     * @version 3.4.19
    **/
    boolean updateLiterature(int literatureId, int status);
}

File: src/main/resources/static/js/service/literatureService.js
/**
 * @author PQ
 * @Description 返回
 * @Date 17:02 20/3/2020
 * @version 3.4.14
**/

/**
 * @author PQ
 * @Description 根据部落获取文章
 * @Date 17:18 20/3/2020
 * @version 3.4.14
**/
function getLiterature(funId, func) {
    $.ajax({
        url: "/api/literature/literature-list",
        type: "GET",
        data: funId,
        success: function (list) {
            func(list);
        },
        error: function (msg) {
            console.log("发送失败");
        }
    })
}

/**
 * @author PQ
 * @Description 根据状态获取文章
 * @Date 17:18 20/3/2020
 * @version 3.4.14
 **/
function getLiteratureByStatus(status, func) {
    $.ajax({
        url: "/api/literature/literature-status-list",
        type: "GET",
        data: status,
        success: function (list) {
            func(list);
        },
        error: function (msg) {
            console.log("发送失败");
        }
    })
}

/**
 * @author PQ
 * @Description 同意文章发表
 * @param data：｛"literatureId": literatureId, "status": status｝
 * @param func: func(boolean)
 * @Date 00:36 21/3/2020
 * @version 3.4.19
 **/
function changeLiteratureStatus(data, func) {
    $.ajax({
        url: "/api/literature/update-status",
        type: "PUT",
        data: data,
        success: function (boolean) {
            func(boolean);
        },
        error: function (msg) {
            console.log("发送失败");
        }
    })
}

/**
 * @author PQ
 * @Description 写文章
 * @param literature: {"title": title, "context": context}
 * @param func: func(boolean)添加成功返回true，否则返回falsze
 * @Date 23:20 23/3/2020
 * @version 3.4.25
**/
function writeLiterature(literature, func) {
    $.ajax({
        url: "/api/literature/write",
        type: "POST",
        data: data,
        success: function (boolean) {
            func(boolean);
        },
        error: function (msg) {
            console.log("发送失败");
        }
    })
}